pennylane/capture/explanations.md

```diff
@@ -425,3 +425,103 @@ Now in our jaxpr, we can see thet `PrimitiveClass2` returns something of type `A
+
+Some of the primitives in the capture module have a somewhat non-standard requirement for the
+behaviour under differentiation or batching: they should ignore that an input is a differentiation
+or batching tracer and just execute the standard implementation on them.
+
+We will look at an example to make the necessity for such a non-interpreted primitive clear.
+
+Consider a finite-difference differentiation routine together with some test function `fun`.
+
+```python
+def finite_diff_impl(x, fun, delta):
+    """Finite difference differentiation routine. Only supports differentiating
+    a function `fun` with a single scalar argument, for simplicity."""
+
+    out_plus = fun(x + delta)
+    out_minus = fun(x - delta)
+    return tuple((out_p - out_m) / (2 * delta) for out_p, out_m in zip(out_plus, out_minus))
+
+def fun(x):
+    return (x**2, 4 * x - 3, x**23)
+```
+
+Now suppose we want to turn this into a primitive. We could just promote it to a standard
+`jax.core.Primitive` as
+
+```python
+import jax
+
+fd_prim = jax.core.Primitive("finite_diff")
+fd_prim.multiple_results = True
+fd_prim.def_impl(finite_diff_impl)
+
+def finite_diff(x, fun, delta=1e-5):
+    return fd_prim.bind(x, fun, delta)
+```
+
+This allows us to use the forward pass as usual (to compute the first-order derivative):
+
+```pycon
+>>> finite_diff(1., fun, delta=1e-6)
+(2.000000000002, 3.999999999892978, 23.000000001216492)
+```
+
+Now if we want to make this primitive differentiable (with automatic
+differentiation/backprop, not by using a higher order finite difference scheme),
+we need to specify a JVP rule. (Note that there are multiple rather simple fixes for this example
+that we could use to implement a finite difference scheme that is readily differentiable. This is
+somewhat besides the point, because we did not identify a possibility to use any of those
+alternatives in the PennyLane code).
+
+However, the finite difference rule is just a standard
+algebraic function making use of calls to `fun` and some elementary operations, so ideally
+we would like to just use the chain rule as it is known to the AD engine. A JVP rule would
+then just manually re-implement this chain rule, which we'd rather not do.
+
+Instead, we define a non-interpreted type of primitives, and create such a primitive
+for our finite difference method. We also create the usual method that binds the
+primitive to inputs.
+
+```python
+class NonInterpPrimitive(jax.core.Primitive):
+    """A subclass to JAX's Primitive that works like a Python function
+    when evaluating JVPTracers."""
+
+    def bind_with_trace(self, trace, args, params):
+        """Bind the ``NonInterpPrimitive`` with a trace.
+        If the trace is a ``JVPTrace``, it falls back to a standard Python function call.
+        Otherwise, the bind call of JAX's standard Primitive is used."""
+        if isinstance(trace, jax.interpreters.ad.JVPTrace):
+            return self.impl(*args, **params)
```

Comment:
What if we have nested tracers? `jax.grad(jax.jit(f))`? Or something like that. 

Though I guess we can just proceed with this till we find an issue.

Reply:
Still seems to work :)